File: PassDataBackwardsObjC/PassDataBackwardsObjC/ViewController.m
//
//  ViewController.m
//  PassDataBackwardsObjC
//
//  Created by Field Employee on 4/2/20.
//  Copyright © 2020 Hugo Flores. All rights reserved.
//

#import "ViewController.h"
#import "SecondaryViewController.h"

@interface ViewController ()

@property (weak, nonatomic) IBOutlet UILabel *myLabel;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (IBAction)onPress:(id)sender {
    SecondaryViewController *vc = [self.storyboard instantiateViewControllerWithIdentifier:@"SecondaryView"];
    vc.mainViewController = self;
    [self presentViewController:vc animated:true completion:nil];
}

- (void)onUserAction:(NSString *)data {
    _myLabel.text = data;
}

@end

File: PassDataBackwardsSwift/PassDataBackwardsSwift/ViewController.swift
//
//  ViewController.swift
//  PassDataBackwardsSwift
//
//  Created by Field Employee on 4/2/20.
//  Copyright © 2020 Hugo Flores. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var dataLabel: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
    }

    @IBAction func onPressed(_ sender: Any) {
        // let vc = SecondaryViewController(nibName: "SecondaryViewControllern", bundle: nil)
        let vc = storyboard?.instantiateViewController(withIdentifier: "SecondaryView") as! SecondaryViewController
        vc.mainViewController = self
        present(vc, animated: true, completion: nil)
    }
    
    func onUserAction(data: String) {
        print("Data received: \(data)")
        dataLabel.text = data
    }
    
}

